app/src/main/java/com/saucelabs/mydemoapp/android/view/activities/MainActivity.java

```diff
@@ -526,4 +546,86 @@ private void showAlert() {
 			.setCancelable(false)
 			.show();
 	}
+
+	/**
+	 * This is a sample function that checks if the current version of this app is still
+	 * available for distribution. If an admin disables distribution or deletes the current
+	 * version, then this function will return (false).
+	 */
+	private boolean isThisVersionStillSupported() {
+		try {
+			TestFairyAssetReader.Data testFairyData = new TestFairyAssetReader().read(this);
+			String appToken = testFairyData.getAppToken();
+
+			PackageInfo packageInfo = getPackageManager().getPackageInfo(getPackageName(), 0);
+
+			URL url = new URL("https://mobile.saucelabs.com/services/?method=testfairy.session.getDistributionStatus");
+			HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+			connection.setRequestMethod("POST");
+			connection.setDoInput(true);
+			connection.setDoOutput(true);
+			connection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+
+			String urlParameters = "token=" + appToken +
+				"&bundleVersion=" + packageInfo.versionCode +
+				"&bundleIdentifier=" + packageInfo.packageName +
+				"&bundleShortVersion=" + packageInfo.versionName +
+				"&platform=0";
+
+			OutputStreamWriter writer = new OutputStreamWriter(connection.getOutputStream());
+			writer.write(urlParameters);
+			writer.flush();
+			writer.close();
+
+			BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+			StringBuffer response = new StringBuffer();
+
+			while (true) {
+				String inputLine = in.readLine();
+				if (inputLine == null) {
+					break;
+				}
+
+				response.append(inputLine);
+			}
+
+			in.close();
+			Log.i(Config.TAG, "Received a response from getDistributionStatus: " + response);
+
+			JSONObject obj = new JSONObject(response.toString());
+			boolean distributionEnabled = "enabled".equals(obj.optString("status"));
+			Log.i(Config.TAG, "Is distribution enabled: " + distributionEnabled);
+			return distributionEnabled;
+
+		} catch (Exception e) {
+			Log.e(Config.TAG, "Failed to check for new beta version with exception", e);
+			return true;
+		}
+	}
+
+	private void showVersionExpiredAlert() {
+		runOnUiThread(new Runnable() {
+			@Override
+			public void run() {
+				new AlertDialog.Builder(mAct, R.style.MyDialogTheme)
+					.setTitle("Version no longer supported")
+					.setMessage("This version has expired. Please upgrade to latest.")
+					.setPositiveButton("Ok", null)
+					.setCancelable(false)
+					.show();
+			}
+		});
+	}
+
+	private void checkVersionIsStillSupported() {
+		new Thread(new Runnable() {
```

Comment:
A small suggestion for threading:

```
ExecutorService executor = Executors.newSingleThreadExecutor();
Handler handler = new Handler(Looper.getMainLooper());

executor.execute(() -> {
    //Background work here
    handler.post(() -> {
        //UI Thread work here
    });
});
```

Reply:
Interesting, updating sample code. Thank you!